sdlib/d/gc/tcache.d

```diff
@@ -199,21 +198,12 @@ public:
 
 		if (pd.isSlab()) {
 			auto si = SlabAllocInfo(pd, slice.ptr);
-
-			size_t freeSize;
-			if (!getSmallFreeSize(slice, si, freeSize)) {
-				return false;
-			}
-
-			if (freeSize < size) {
-				return false;
-			}
-
-			si.setFreeSpace(freeSize - size);
-			return true;
+			auto usedCapacity = si.usedCapacity;
+			return validateCapacity(slice, si.address, usedCapacity)
```

Comment:
That'd be clearer with an early return, especially since we count on the short circuit to kick in.